src/amgx_cusparse.cu

```diff
@@ -1008,55 +1053,70 @@ inline void generic_SpMV(cusparseHandle_t handle, cusparseOperation_t trans,
                              cudaDataType vecType,
                              const cudaStream_t& stream)
 {
-    int col_off;
-    cudaMemcpyAsync(&col_off, &rowPtr[0], sizeof(int), cudaMemcpyDefault, stream);
-    cudaStreamSynchronize(stream);
-
-    IndType* rows = const_cast<IndType*>(rowPtr);
-    IndType* cols = const_cast<IndType*>(colInd) + col_off;
-    MatType* vals = const_cast<MatType*>(val) + col_off;
+    constexpr int cta_size = 128;
+    const int sm_count = thrust::global_thread_handle::get_sm_count();
```

Comment:
Another minor thing - it seems weird to get _SM count_ from something called _global thread handle_ from namespace _thrust_ ;) This can be refactored later.

Reply:
Actually this is a good point - this really should just be in `src/device_properties.cu`. I'll move it now.